### PR TITLE
Feat: Late type, expects a type that gets defined later.

### PR DESCRIPTION
**Pull request by mattiamanzati**
#74

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import "./types/complex-types/complex-type"
 export {
     types
 } from "./types"
+export {
+    IType
```
Comment (mweststrate) [inline src/index.ts]: just `IType` on line 6 should work as well

File: src/types/utility-types/recursive.ts [hunk 2]
```diff
@@ -1,28 +0,0 @@
-import { IType, Type } from "../type"
```
Comment (mweststrate) [inline src/types/utility-types/recursive.ts]: I justed wanted to ask you about that :). Seems you are reading my mind...

File: src/index.ts (hunk 1 above)
Comment (mattiamanzati) [inline src/index.ts]: Unfortunately "./types" does not exports IType, so should be the same as requiring it here :)

File: src/types/utility-types/recursive.ts (hunk 2 above)
Comment (mattiamanzati) [inline src/types/utility-types/recursive.ts]: Yeah, only reason to keep that would be to provide a better describe() in circulars, but that's not bat at all as it works now.

**Pull request merged**